system/lib/fetch/emscripten_fetch.cpp

```diff
@@ -144,8 +177,26 @@ EMSCRIPTEN_RESULT emscripten_fetch_close(emscripten_fetch_t *fetch)
 		strcpy(fetch->statusText, "aborted with emscripten_fetch_close()");
 		fetch->__attributes.onerror(fetch);
 	}
+
+	emscripten_fetch_free(fetch);
+	return EMSCRIPTEN_RESULT_SUCCESS;
+}
+
+void emscripten_fetch_free(emscripten_fetch_t *fetch)
```

Comment:
Good idea to share the implementation out here, but to make this private, let's rename to `static void _emscripten_fetch_free(emscripten_fetch_t *fetch)` so that it is local only to this cpp file?